blobfuse/fileapis.cpp

```diff
@@ -12,6 +12,53 @@ std::deque<file_to_delete> cleanup;
 std::mutex deque_lock;
 std::shared_ptr<gc_cache> g_gc_cache;
 
+int DownloadFileToDisk(std::string pathString, std::string mntPathString, BfsFileProperty blob_property, bool is_delayed)
+{
+    errno = 0;
+    time_t last_modified = {};
+
+    std::string disk_path = mntPathString;
+
+    syslog(LOG_DEBUG, "Starting download of file  %s", pathString.c_str());
+    long int size = storage_client->DownloadToFile(pathString.substr(1), disk_path, last_modified);
+    if (errno != 0)
+    {
+        int storage_errno = errno;
+        syslog(LOG_ERR, "Failed to download blob into cache.  Blob name: %s, file name = %s, storage errno = %d.\n", pathString.c_str()+1, mntPathString.c_str(),  errno);
+
+        remove(mntPathString.c_str());
+        return 0 - map_errno(storage_errno);
+    }
+    else
+    {
+        syslog(LOG_INFO, "Successfully downloaded blob %s into file cache as %s.\n", pathString.c_str()+1, mntPathString.c_str());
+        g_gc_cache->addCacheBytes(mntPathString, size);
+    }
+
+    // preserve the last modified time
+    struct utimbuf new_time;
```

Comment:
these 3 lines are not required. It retains the last modified time anyway

Reply:
We want last modified time to reflect the one mentioned in container. When we create a local file and dump data into it system will change time to current time, hence post dumping the data we need to reset the last modified time back to one mentioned in container. We were doing the same earlier also just that download of file logic has been moved to another function now.